FORMAT_FAILURE encountered in Camel logging from dev mode

Originally reported for Camel Quarkus here:

https://github.com/apache/camel-quarkus/issues/7843

There's a reproducer project here (run with `mvn clean quarkus:dev`):

https://github.com/jam01/camel-log-repro/tree/main

When an exception is thrown from within Camel route processing, Camel will include some extended information about the message exchange in the generated log message.

Camel wraps the message exchange body within the log message between `{}` (not entirely sure why TBH). This seems to break JBoss logging formatting.

E.g if the message body is a string like `foo`, then you get a log message string like:

```
Message History (source location is disabled)
---------------------------------------------------------------------------------------------------------------------------------------
Source                                   ID                             Processor                                          Elapsed (ms)
                                         route1/route1                  from[timer://foo?period=1s]                                  12
                                         route1/setBody1                setBody[constant{foo}]                                        2
```

In dev mode it results in (due to the `setBody[constant{foo}]`):

```
LogManager error of type FORMAT_FAILURE: Formatting error
java.lang.IllegalArgumentException: can't parse argument number: foo
	at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1479)
	at java.base/java.text.MessageFormat.applyPattern(MessageFormat.java:511)
	at java.base/java.text.MessageFormat.<init>(MessageFormat.java:382)
	at java.base/java.text.MessageFormat.format(MessageFormat.java:882)
	at org.jboss.logmanager.ExtFormatter.formatMessageLegacy(ExtFormatter.java:133)
	at org.jboss.logmanager.ExtFormatter.formatMessage(ExtFormatter.java:96)
	at org.jboss.logmanager.formatters.Formatters$16.renderRaw(Formatters.java:832)
	at org.jboss.logmanager.formatters.Formatters$JustifyingFormatStep.render(Formatters.java:227)
	at org.jboss.logmanager.formatters.MultistepFormatter.format(MultistepFormatter.java:90)
	at org.jboss.logmanager.ExtFormatter.format(ExtFormatter.java:58)
	at org.jboss.logmanager.handlers.WriterHandler.doPublish(WriterHandler.java:54)
	at org.jboss.logmanager.ExtHandler.publish(ExtHandler.java:88)
	at org.jboss.logmanager.ExtHandler.publishToNestedHandlers(ExtHandler.java:125)
	at io.quarkus.bootstrap.logging.QuarkusDelayedHandler.doPublish(QuarkusDelayedHandler.java:81)
	at org.jboss.logmanager.ExtHandler.publish(ExtHandler.java:88)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:437)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:479)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:479)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:479)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:479)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:479)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:479)
	at org.jboss.logmanager.Logger.logRaw(Logger.java:921)
	at org.slf4j.impl.Slf4jLogger.log(Slf4jLogger.java:574)
	at org.slf4j.impl.Slf4jLogger.error(Slf4jLogger.java:515)
	at org.apache.camel.spi.CamelLogger.log(CamelLogger.java:205)
	at org.apache.camel.spi.CamelLogger.log(CamelLogger.java:222)
	at org.apache.camel.spi.CamelLogger.log(CamelLogger.java:113)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.logFailedDelivery(RedeliveryErrorHandler.java:631)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.prepareExchangeAfterFailureNotHandled(RedeliveryErrorHandler.java:578)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.prepareExchangeAfterFailure(RedeliveryErrorHandler.java:557)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.handlePreviousFailure(RedeliveryErrorHandler.java:446)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.run(RedeliveryErrorHandler.java:413)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.done(RedeliveryErrorHandler.java:392)
	at org.apache.camel.processor.ThrowExceptionProcessor.process(ThrowExceptionProcessor.java:82)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.handleFirst(RedeliveryErrorHandler.java:440)
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.run(RedeliveryErrorHandler.java:416)
	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.doRun(DefaultReactiveExecutor.java:199)
	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.executeReactiveWork(DefaultReactiveExecutor.java:189)
	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.tryExecuteReactiveWork(DefaultReactiveExecutor.java:166)
	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:148)
	at org.apache.camel.impl.engine.DefaultReactiveExecutor.scheduleMain(DefaultReactiveExecutor.java:59)
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:162)
	at org.apache.camel.impl.engine.CamelInternalProcessor.processNonTransacted(CamelInternalProcessor.java:370)
	at org.apache.camel.impl.engine.CamelInternalProcessor.process(CamelInternalProcessor.java:346)
	at org.apache.camel.component.timer.TimerConsumer.sendTimerExchange(TimerConsumer.java:293)
	at org.apache.camel.component.timer.TimerConsumer$1.doRun(TimerConsumer.java:164)
	at org.apache.camel.component.timer.TimerConsumer$1.run(TimerConsumer.java:136)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
	at java.base/java.util.TimerThread.run(Timer.java:516)
Caused by: java.lang.NumberFormatException: For input string: "foo"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Integer.parseInt(Integer.java:662)
``` 

In prod mode, things seem to be working ok. There's no exception visible in the logs emanating from JBoss Logging.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -106,7 +106,7 @@
         <wildfly.openssl-java.version>2.2.5.Final</wildfly.openssl-java.version>
         <wildfly.openssl-linux.version>2.2.2.SP01</wildfly.openssl-linux.version>
         <jboss-logging-annotations.version>3.0.4.Final</jboss-logging-annotations.version>
-        <slf4j-jboss-logmanager.version>2.0.0.Final</slf4j-jboss-logmanager.version>
+        <slf4j-jboss-logmanager.version>2.0.2.Final</slf4j-jboss-logmanager.version>
         <wildfly-common.version>2.0.1</wildfly-common.version>
         <wildfly-client-config.version>1.0.1.Final</wildfly-client-config.version>
         <wildfly-elytron.version>2.7.0.Final</wildfly-elytron.version>

